openedx_tagging/core/tagging/models.py

```diff
@@ -409,6 +413,32 @@ def tag_object(
 
         return updated_tags
 
+    def autocomplete_tags(self, search: str, object_id: str = None) -> models.QuerySet:
+        """
+        Returns tag values that contains the given search string.
+        The result is returned in alphabetical order.
+
+        The output is a QuerySet of dictionaries in with `_value` and `tag`.
+
+        Subclasses can override this method to perform their own autocomplete process.
+        """
+        # Fetch tags that the object already has to exclude them from the result
+        excluded_tags = self.objecttag_set.filter(object_id=object_id).values_list(
+            "_value", flat=True
+        )
```

Comment:
If no object_id is provided, we don't need to make this call..  so can this be:

```suggestion
        excluded_tags = []
        if object_id:
            excluded_tags = self.objecttag_set.filter(object_id=object_id).values_list(
                "_value", flat=True
            )
```

Reply:
Updated here: https://github.com/openedx/openedx-learning/pull/59/commits/593c047b8d7bf5c3a314b8a25d19f55beb5238c8